ansible_builder/cli.py

```diff
@@ -34,8 +50,10 @@ def run():
     ab = prepare()
 
     print('Processing...', end='\r')
+    # TO DO: stdout that prints in real-time and/or a "spinner"
 
-    if ab.process():
+    build_or_create = getattr(ab, sys.argv[1])
```

Comment:
this use of `sys.argv` undermines the role of argparse.

The only way to solve this is to make the action one of the primary parameters in this main data bus. Consider:

```diff
diff --git a/ansible_builder/cli.py b/ansible_builder/cli.py
index 77b1571..eda4f36 100644
--- a/ansible_builder/cli.py
+++ b/ansible_builder/cli.py
@@ -7,7 +7,7 @@ from . import constants
 
 def prepare(args=sys.argv[1:]):
     parser = argparse.ArgumentParser(prog='ansible-builder')
-    subparsers = parser.add_subparsers(help='The command to invoke.')
+    subparsers = parser.add_subparsers(help='The command to invoke.', dest='action')
 
     create_command_parser = subparsers.add_parser('create',
                                                   help='Outputs a build context, including a Containerfile, populated with dependencies.')
@@ -33,7 +33,7 @@ def prepare(args=sys.argv[1:]):
                        help='The parent image for the execution environment.')
 
         p.add_argument('-c', '--context',
-                       default=None,
+                       default=constants.default_build_context,
                        dest='build_context',
                        help='The directory to use for the build context. Defaults to $PWD/context.')
 
@@ -52,7 +52,7 @@ def run():
     print('Processing...', end='\r')
     # TO DO: stdout that prints in real-time and/or a "spinner"
 
-    build_or_create = getattr(ab, sys.argv[1])
+    build_or_create = getattr(ab, ab.action)
     if build_or_create():
         print("Complete! Build context is at: {}".format(ab.build_context))
         sys.exit(0)
diff --git a/ansible_builder/main.py b/ansible_builder/main.py
index 33e1285..bcc9068 100644
--- a/ansible_builder/main.py
+++ b/ansible_builder/main.py
@@ -8,12 +8,13 @@ from . import constants
 
 
 class AnsibleBuilder:
-    def __init__(self,
+    def __init__(self, action=None,
                  filename=constants.default_file,
                  base_image=constants.default_base_image,
                  build_context=constants.default_build_context,
                  tag=constants.default_tag,
                  container_runtime=constants.default_container_runtime):
+        self.action = action
         self.definition = Definition(filename=filename)
         self.base_image = base_image
         self.tag = tag
@@ -78,7 +79,10 @@ class Definition:
 class Containerfile:
     newline_char = '\n'
 
-    def __init__(self, *args, filename, definition, build_context, base_image):
+    def __init__(self, definition,
+                 filename=constants.default_file,
+                 build_context=constants.default_build_context,
+                 base_image=constants.default_base_image):
         self.build_context = build_context
         os.makedirs(self.build_context, exist_ok=True)
         self.definition = definition
```